src/twisted/application/runner/test/test_runner.py

```diff
@@ -68,11 +68,31 @@ def test_runInOrder(self):
         """
         L{Runner.run} calls the expected methods in order.
         """
-        runner = DummyRunner({})
+
+        calledMethods = []
+
+        self.patch(
```

Comment:
Bad as inheritance is, global patching seems worse.  Why didn't you like `DummyRunner` as an approach to testing this?